AndroidTest/app/src/main/java/cloudant/com/androidtest/keyprovider/AndroidKeyProviderTests.java

```diff
@@ -0,0 +1,173 @@
+package cloudant.com.androidtest.keyprovider;
+
+import android.test.AndroidTestCase;
+
+import com.cloudant.sync.datastore.encryption.AndroidKeyProvider;
+import com.cloudant.sync.datastore.encryption.DPKException;
+import com.cloudant.sync.datastore.encryption.EncryptionKey;
+import com.cloudant.sync.datastore.encryption.KeyManager;
+import com.cloudant.sync.datastore.encryption.KeyStorage;
+
+import java.util.Arrays;
+
+public class AndroidKeyProviderTests extends AndroidTestCase {
+
+    public void testCreateProviderWithIdentifier() {
+        AndroidKeyProvider provider = new AndroidKeyProvider(getContext(), "provider1password",
+                ProviderTestUtil.getUniqueIdentifier());
+
+        EncryptionKey createdKey = provider.getEncryptionKey();
+        assertNotNull("EncryptionKey was not created", createdKey);
+
+        EncryptionKey loadedKey = provider.getEncryptionKey();
+        assertNotNull("EncryptionKey was not loaded", loadedKey);
+
+        assertTrue("loadedKey did not match createdKey", Arrays.equals(createdKey.getKey(),
+                loadedKey.getKey()));
```

Comment:
Correct. Even though the method is called twice, different things happen each time.  The first call generates the key and stores it.  The DPK return is never encrypted while the stored DPK is encrypted.  The second call to getEncryptionkey does a load.  Load will need to decrypt the key before returning it.  This test validates that the create key equals the decrypted loaded key.